File: src/main/java/com/syncstream/syncstream/model/user/UserInfo.java
package com.syncstream.syncstream.model.user;

import java.time.LocalDateTime;

public class UserInfo {
    private String token;
    private String email;
    private String password;
    private LocalDateTime creationDate;
    private LocalDateTime lastLogin;
    private String avatarUrl;

    public UserInfo(String token, String email, String password, LocalDateTime creationDate, LocalDateTime lastLogin, String avatarUrl) {
        this.token = token;
        this.email = email;
        this.password = password;
        this.creationDate = creationDate;
        this.lastLogin = lastLogin;
        this.avatarUrl = avatarUrl;
    }

    public String getToken() {
        return token;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public LocalDateTime getCreationDate() {
        return creationDate;
    }

    public LocalDateTime getLastLogin() {
        return lastLogin;
    }

    public String getAvatarUrl() {
        return avatarUrl;
    }
}

File: syncstream/src/datamodules/user/user_info.rs
// user_info.rs implements the UserInfo module
use chrono::offset::Local;
use chrono::DateTime;

pub struct UserInfo {
    token: String,
    email: String,
    password: String,
    creation_date: DateTiem<Local>,
    last_login: DateTime<Local>,
    avatar_url: String
}

impl UserInfo {
    pub fn new(token: String,
        email: String,
        password: String,
        creation_date: DateTime<Local>,
        last_login: DateTime<Local>,
        avatar_url: String,
    ) -> Self {
        UserInfo {
            token,
            email,
            password,
            creation_date,
            last_login,
            avatar_url,
        }
    }
}
